### ISSUE TITLE
UI: Change behavior of address fields in volunteer and admin sign-up

### ISSUE THREAD

**Issue by rkpattnaik780** [anitab-org/vms#1050]
## Description
As a volunteer/admin, while signing up, the form fields pertaining to address should be enabled only if the dependent fields are filled.
The "state/province" field should be enabled only after country is selected. City should be enabled only if both "country" and "state/province" are selected.

### Update [Required]
- [ ] GIF of the behavior of form fields.
- [ ] Changed behavior of form fields.

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2 hours

**Comment by ymaheshwari1** [comment]
I am interested in working on this.

**Comment by abhinav72610** [comment]
@rkpattnaik780 Hey I am new in this community and I am participating in OSH can I take this up?

**Comment by yash2189** [comment]
Can I work on this @rkpattnaik780 ?

**Comment by psingh587** [comment]
Hi, I am interested to work on this. Can I do it?

**Comment by rkpattnaik780** [comment]
> Hi, I am interested to work on this. Can I do it?

@psingh587 go ahead!

**Comment by psingh587** [comment]
Sure, Thanks!

**Comment by rkpattnaik780** [comment]
@psingh587 are you still working on this issue?

**Comment by psingh587** [comment]
Sir, I am having problems in installation of VMS in my machine. I had asked for help yesterday but no response yet. Could you please help me? I have checked the files on the repo, i can fix this, only problem is the installation.

**Comment by rkpattnaik780** [comment]
Let me try to run vms using docker compose, I will reach back after that.

**Comment by psingh587** [comment]
My Installation is done. I am working on this. Thanks so much for your help

**Comment by psingh587** [comment]
Sir, I have a question.
![Screenshot from 2020-09-05 18-21-53](https://user-images.githubusercontent.com/51945896/92305423-dbd3fe80-efa4-11ea-92c9-ddc557e62573.png)


This is what I see in volunteer sign up page. In this, City is there first, then state, then country. But in live vms, its different. 
![Screenshot from 2020-09-05 18-24-36](https://user-images.githubusercontent.com/51945896/92305442-1473d800-efa5-11ea-99fe-26a54e8f7749.png)

Here its written in order. Where am I going wrong? Also, there is no dropdown for city, country as well. another issue has already been raised on it.

**Comment by psingh587** [comment]
@rkpattnaik780 Anything on it?

**Comment by rkpattnaik780** [comment]
@psingh587 sorry for the delayed response. I can see the change can be done only to admin signup page, you just need to disable the respective dropdowns with something like `<select disabled>`. You have to conditionally put the disabled attribute.

**Comment by psingh587** [comment]
Okay so that means I only have to do it in the admin sign up page right? 
I have written a JS function and its working as well. I will attach it.
I have written that function in another JS file which was already there containing other function. Should I make a new file? or Will it be okay if I write it in this one? Also , when I am trying to make another file , I cant see the change I made. 
How shall I proceed?
![WhatsApp-Video-2020-09-06-at-173](https://user-images.githubusercontent.com/51945896/92325377-ae03be00-f067-11ea-8e28-cb0d9213744c.gif)

**Comment by rkpattnaik780** [comment]
Hello @psingh587 . Yup just the admin signup.
I was thinking this wouldn't be needing a JS file. You can conditionally add an attribute to the template. I googled it right now, you can do something like [this](https://stackoverflow.com/questions/28644951/how-to-apply-conditional-attribute-to-html-element-in-template).
And this is bit out of context, to easily record screen, use [peek](https://github.com/phw/peek) :)

**Comment by psingh587** [comment]
> Hello @psingh587 . Yup just the admin signup.
> I was thinking this wouldn't be needing a JS file. You can conditionally add an attribute to the template. I googled it right now, you can do something like [this](https://stackoverflow.com/questions/28644951/how-to-apply-conditional-attribute-to-html-element-in-template).
> And this is bit out of context, to easily record screen, use [peek](https://github.com/phw/peek) :)

Hi,
Actually I had thought about it earlier- to do it inside the HTML only. But I am not able to figure out how to write the condition to disable=" " . How to get the value of country dynamically? Please help me here.
<a href="https://stackoverflow.com/questions/32875751/display-form-field-based-on-other-fields-within-the-django-form">I checked it here as well</a>

**Comment by rkpattnaik780** [comment]
> > Hello @psingh587 . Yup just the admin signup.
> > I was thinking this wouldn't be needing a JS file. You can conditionally add an attribute to the template. I googled it right now, you can do something like [this](https://stackoverflow.com/questions/28644951/how-to-apply-conditional-attribute-to-html-element-in-template).
> > And this is bit out of context, to easily record screen, use [peek](https://github.com/phw/peek) :)
> 
> Hi,
> Actually I had thought about it earlier- to do it inside the HTML only. But I am not able to figure out how to write the condition to disable=" " . How to get the value of country dynamically? Please help me here.
> [I checked it here as well](https://stackoverflow.com/questions/32875751/display-form-field-based-on-other-fields-within-the-django-form)

It seems bit tricky to do it in htmll. You can handle it through javascript, won't be an issue. Just  handle all the scenarios.

**Comment by rkpattnaik780** [comment]
There is no necessity to create a new js file, `load_city.js` and `load_state.js` should do the work.

**Comment by psingh587** [comment]
> There is no necessity to create a new js file, `load_city.js` and `load_state.js` should do the work.

Okay, So I am supposed to make the changes in these files only right?

**Comment by psingh587** [comment]
> There is no necessity to create a new js file, `load_city.js` and `load_state.js` should do the work.

Working on it!

**Comment by psingh587** [comment]
@rkpattnaik780  Please look into my PR
